pyjuque/Engine/Models_mysql.py

```diff
@@ -0,0 +1,152 @@
+
+  def getPairs(self, session):
+    return session.query(Pair).filter_by(bot_id=self.id).all()
+
+  def getActivePairs(self, session):
+    return session.query(Pair).filter_by(bot_id=self.id, active=True).all()
+
+  def getPairWithSymbol(self, session, symbol):
+    return session.query(Pair).filter_by(bot_id=self.id, symbol=symbol).first()
+
+  def getOrders(self, session):
+    return session.query(Order).filter_by(bot_id=self.id).all()
+
+  def getOpenOrders(self, session):
+    return session.query(Order).filter_by(bot_id=self.id, is_closed=False).all()    
+
+  def getFirstBuyOrder(self, session, position_id):
+    return session.query(Order).filter_by(bot_id=self.id, position_id=position_id, side='BUY').first()
+    
+class EntrySettings(Base):
+  """
+  Model for entry settings
+  """
+
+  __tablename__ = 'entry_settings'
+  id = db.Column(db.Integer, primary_key=True)                  	# Unique ID
+  name = db.Column(db.String(30))                                 # Name (for UI)
+  bots = relationship('Bot', backref=backref('entry_settings'))
+  open_buy_order_time_out = db.Column(db.Integer, default=0)
+  initial_entry_allocation = db.Column(db.Integer, default=None)	# What % of funds allocated to the bot will go to an initial entry
+  subsequent_entries = db.Column(db.Integer, default=0)         	# Are there subsequent entries
+  subsequent_entry_allocation = db.Column(db.Float, default=1)  	# What % of the initial quantity will we buy on a 
+                                  # subsequent entry (1 = 100%, 0.5 = 50%)
+  subsequent_entry_distance = db.Column(db.Float, default=None) 	# Distance between subsequent entries in %
+  signal_distance = db.Column(db.Float, default=0)         		# Distance from signal for initial entry 
+                                    # 0 means enter on signal, 1 means enter 1% away 
+                                    # - in the direction opposite that of the trade - 
+                                    # from the signal price
+
+class ExitSettings(Base):
+    """ Exit Settings of a Bot """
+    __tablename__ = 'exit_settings'
+    id = db.Column(db.Integer, primary_key=True)                  	# Unique ID
+    name = db.Column(db.String(30))                               	# Name for UI
+    bots = relationship('Bot', backref=backref('exit_settings'))   # Name (for UI)
+    profit_target = db.Column(db.Float)                           	# Exit when price is at value % profit from entry 
+    stop_loss_value = db.Column(db.Float, default=None)             	# Whether to have stop loss or not (and what %)
+    is_trailing_stop_loss = db.Column(db.Boolean, default=False)    	# Whether to have trailing stop loss or not (what %)
+    stop_loss_active_after = db.Column(db.Float, default=None)    	# If we have trailing stop loss, whether to activate immediately, or after a value % increase in profit
+    exit_on_signal = db.Column(db.Boolean, default=False)       
```

Comment:
How is this changed compared to the normal Models?

Reply:
The point of using SQLAlchemy is that we can use any kind of sql DB we want - sqlite, mysql, psql